src/main/cpp/drivers/mock_io.cpp

```diff
@@ -1,39 +1,41 @@
-#include "mock_opensl_io.h"
-#include "../../audioengine.h"
-#include "../../sequencer.h"
-#include "../../utilities/debug.h"
-
-inline OPENSL_STREAM* mock_android_OpenAudioDevice( int sr, int inchannels, int outchannels, int bufferframes )
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2020 Igor Zinken - https://www.igorski.nl
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "mock_io.h"
+#include "../audioengine.h"
+#include "../sequencer.h"
+#include <utilities/debug.h>
+
+Mock_IO::Mock_IO()
 {
-    Debug::log( "mocked device opening" );
-
-    OPENSL_STREAM *p = ( OPENSL_STREAM* ) calloc( sizeof( OPENSL_STREAM ), 1 );
-
-    p->inchannels  = inchannels;
-    p->outchannels = outchannels;
-    p->sr = sr;
-
-    AudioEngine::engine_started = true;
-
-    return p;
```

Comment:
With the adapter mechanism there is no need for the mocked driver to rely on OpenSL and stubbing its internal methods.